lib/experimental/fonts-library/class-wp-fonts-library.php

```diff
@@ -0,0 +1,618 @@
+     */
+    function delete_font_face_assets ( $font_face ) {
+        if ( is_array( $font_face['src'] ) ) {
+            foreach ( $font_face['src'] as $src ) {
+                $was_assets_removed = $this->delete_asset($src);
+                if ( ! $was_assets_removed ) {
+                    // Bail if any of the assets could not be removed
+                    return false;
+                }
+            }
+            return true;
+        } else {
+            return $this->delete_asset( $font_face['src'] );
+        }
+    }
+
+    /**
+     * Downloads a font asset.
+     *
+     * Downloads a font asset from a specified source URL and saves it to the font directory.
+     *
+     * @param string $src The source URL of the font asset to be downloaded.
+     * @param string $filename The filename to save the downloaded font asset as.
+     * @return string|bool The relative path to the downloaded font asset. False if the download failed.
+     */
+    function download_asset( $src, $filename ) {
+        $file_path = path_join( $this->wp_fonts_dir, $filename );
+
+        // Checks if the file to be downloaded has a font mime type
+        if ( ! $this->has_font_mime_type( $file_path ) ) {
+            return false;
+        }
+
+        // Downloads the font asset or returns false
+        $temp_file = download_url( $src );
+        if ( is_wp_error( $temp_file ) ) {
+            @unlink( $temp_file );
+            return false;
+        }
+
+        // Moves the file to the fonts directory or return false
+        $renamed_file = rename( $temp_file, $file_path );
+        if ( ! $renamed_file ) {
+            @unlink( $temp_file );
+            return false;
+        }
+
+        // Cleans the temp file
+        @unlink( $temp_file );
+
+        // Returns the relative path to the downloaded font asset to be used as font face src
+        return "{$this->relative_fonts_path}{$filename}";
+    }
+
+    /**
+     * Generates a filename for a font face asset.
+     *
+     * Creates a filename for a font face asset using font family, style, weight and extension information.
+     *
+     * @param array $font_face The font face array containing 'fontFamily', 'fontStyle', and 'fontWeight' attributes.
+     * @param string $url The URL of the font face asset, used to derive the file extension.
+     * @param int $i Optional counter for appending to the filename, default is 0.
+     * @return string The generated filename for the font face asset.
+     */
+    function get_filename_from_font_face ( $font_face, $url, $i=0 ) {
+        $extension = pathinfo( $url, PATHINFO_EXTENSION );
+        $family = sanitize_title( $font_face['fontFamily'] );
+        $style = sanitize_title( $font_face['fontStyle'] );
+        $weight = sanitize_title( $font_face['fontWeight'] );
+        $filename = "{$family}_{$style}_{$weight}";
+        if ($i > 0) {
+            $filename = "{$filename}_{$i}";
```

Comment:
This could be replaced with
```suggestion
            $filename .= "_{$i}";
```